connectors/sources/confluence.py

```diff
@@ -258,9 +279,18 @@ async def paginated_api_call(self, url_name, **url_kwargs):
                         self.host_url,
                         links.get("next")[1:],
                     )
+            except ServerDisconnectedError:
+                raise
+            except ClientResponseError:
+                raise
+            except ClientPayloadError as exception:
+                self._logger.error(
+                    f"Payload error occurred while fetching {url_name} from {url}. Exception: {exception}."
+                )
+                break
             except Exception as exception:
```

Comment:
This looks like we'll still be swallowing _most_ exceptions that occur while paginating. Can you explain why not switch to something like:
```
except ErrorWeCanSafelyIgnore:
  self._logger.warn("skipping paginating the rest, because of {good reason}")
except Exception:
  self._logger.error("We hit an unexpected error while paginating, and that could be a problem you want to handle")
  raise
```